cpp/src/IceGrid/Internal.ice

```diff
@@ -13,420 +13,418 @@
+        ["cpp:const"] idempotent long getOffsetFromEnd(string filename, int lines)
+            throws FileNotAvailableException;
+
+        /// Read lines (or size bytes) at the specified position from the given file.
+        ["cpp:const"] idempotent bool read(string filename, long pos, int size, out long newPos, out Ice::StringSeq lines)
+            throws FileNotAvailableException;
+    }
+
+    interface Server extends FileReader
+    {
+        /// Start the server.
+        ["amd"] void start()
+            throws ServerStartException;
+
+        /// Stop the server. This methods returns only when the server is deactivated. If the server doesn't stop after a
+        /// configurable amount of time, it will be killed.
+        ["amd"] void stop()
+            throws ServerStopException;
+
+        /// Check if the given server can be loaded on this node.
+        /// @return True if the server is inactive.
+        /// @throws DeploymentException Raised if the server can't be updated.
+        bool checkUpdate(InternalServerDescriptor svr, bool noRestart)
+            throws DeploymentException;
+
+        /// Enable or disable the server.
+        void setEnabled(bool enable);
+
+        /// Check if the server is enabled.
+        ["cpp:const"] idempotent bool isEnabled();
+
+        /// Send signal to the server
+        void sendSignal(string signal)
+            throws BadSignalException;
+
+        /// Write message on servers' stdout or stderr.
+        void writeMessage(string message, int fd);
+
+        /// Return the server state.
+        /// @return The server state.
+        /// @see ServerState
+        ["cpp:const"] idempotent ServerState getState();
+
+        /// Get the server pid. Note that the value returned by this method is system dependant. On Unix operating systems,
+        /// it's the pid value returned by the fork() system call and converted to an integer.
+        ["cpp:const"] idempotent int getPid();
+
+        /// Set the process proxy.
+        ["amd"] void setProcess(Ice::Process* proc);
+    }
+
+    interface InternalRegistry;
+    sequence<InternalRegistry*> InternalRegistryPrxSeq;
+
+    interface ReplicaObserver
+    {
+        /// Initialization of the replica observer.
+        void replicaInit(InternalRegistryPrxSeq replicas);
+
+        /// Notification that a replica has been added. The node should establish a session with this new replica.
+        void replicaAdded(InternalRegistry* replica);
+
+        /// Notification that a replica has been removed. The node should destroy the session to this replica.
+        void replicaRemoved(InternalRegistry* replica);
+    }
+
+    interface Node extends FileReader, ReplicaObserver
+    {
+        /// Load the given server. If the server resources weren't already created (database environment directories,
+            /// property files, etc), they will be created. The returned proxy is never null.
```

Comment:
The 2nd line got indented. Possible because it contains '(' and ')'?
Not worth addressing in the script, but still should be fixed:
```suggestion
        /// Load the given server. If the server resources weren't already created (database environment directories,
        /// property files, etc), they will be created. The returned proxy is never null.
```